core/src/main/java/com/linecorp/armeria/server/ServerConfig.java

```diff
@@ -169,6 +177,13 @@
                                    gracefulShutdownQuietPeriod, "gracefulShutdownQuietPeriod");
 
         requireNonNull(blockingTaskExecutor, "blockingTaskExecutor");
+        if (blockingTaskExecutor instanceof BlockingTaskExecutor) {
+            blockingTaskExecutor = ((BlockingTaskExecutor) blockingTaskExecutor).unwrap();
```

Comment:
We should only use the unwrapped executor for `ExecutorServiceMetrics.monitor()`.
Please pass the original blockingTaskExecutor to UnstoppableScheduledExecutorService.from()`

Reply:
We override `blokingTaskExecutor` by the return value of `ExecutorServiceMetircs.monitor()` and then pass it to the `UnstoppableScheduledExecutorService.from()`.
Therefore, I think it's not necessary to save the original value separately and use it.

If I was wrong, please correct me :)

Reply:
A user would find it surprising if the user specified his/her own `BlockingTaskExecutor` implementation. We have to keep the original implementation and use the unwrapped instance only for monitoring.

Reply:
But `ExecutorServiceMetrics#monitor(...)` returns proxied `TimedScheduledExecutorService`.
We may loose some timer metrics if we choose to pass the wrapped `BlockingTaskExecutor` as is.

Reply:
Uh, indeed.. :cry: How about this compromise?

```java
if (blockingTaskExecutor instanceof BlockingTaskExecutor) {
  if (blockingTaskExecutor instanceof DefaultBlockingTaskExecutor) {
    .. Construct a new DefaultBlockingTaskExecutor with monitored ScheduledExecutorService ..
  } else {
    .. Don't monitor ..
  }
}
```

Reply:
Otherwise, as a hacky way, we can manually call https://github.com/micrometer-metrics/micrometer/blob/1e797415edb110e01214257e035c3d1b37b7e1e2/micrometer-core/src/main/java/io/micrometer/core/instrument/binder/jvm/ExecutorServiceMetrics.java#L173
```java
ExecutorServiceMetrics.monitor(((BlockingTaskExecutor) blockingTaskExecutor).unwrap(), ...)
blockingTaskExecutor = new TimedScheduledExecutorService(registry, blockingTaskExecutor, "blockingTaskExecutor", "armeria.", ImmutableList.of());
this.blockingTaskExecutor = UnstoppableScheduledExecutorService.from(blockingTaskExecutor);
```

Reply:
I apply it. Please check again 😄